### PR TITLE
Bug 1878641: Add View only access form for Health checks

### PR DESCRIPTION
**Pull request by sahil143**
**Fixes**: https://issues.redhat.com/browse/ODC-3813

**Analysis / Root cause**: Missing view access on the Health check forms

**Solution Description**: Added view Access for Health check forms

**Screen shots / Gifs for design review**: 

Screenshot 2020-09-14 at 2 43 02 PM
Screenshot 2020-09-14 at 2 43 20 PM
Screenshot 2020-09-14 at 2 43 38 PM
Screenshot 2020-09-14 at 2 43 56 PM
Screenshot 2020-09-14 at 2 45 23 PM

**Unit test coverage report**: 

**Test setup:**

**Browser conformance**: 

- [x] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge

### CODE REVIEW

File: frontend/packages/console-app/src/actions/modify-health-checks.ts [hunk 1]
```diff
@@ -38,15 +38,15 @@ export const AddHealthChecks = (model: K8sKind, obj: K8sResourceKind): KebabOpti
 
 export const EditHealthChecks = (model: K8sKind, obj: K8sResourceKind): KebabOption => {
   return {
-    label: 'Edit Health Checks',
+    label: `Edit Health Checks`,
```
Comment (divyanshiGupta) [inline frontend/packages/console-app/src/actions/modify-health-checks.ts:41]: No need of using string literals here

**Comment by invincibleJai** [comment]
Verified and Overall looks good to me , just need to test scenarios with `ContainerCommand` and `TCPSocket` for Request type.

File: frontend/packages/dev-console/src/components/health-checks/HealthChecksProbe.tsx [hunk 2]
```diff
@@ -59,22 +61,28 @@ const HealthCheckProbe: React.FC<HealthCheckProbeProps> = ({ probeType }) => {
...
-          <Tooltip content="Remove" position="right">
-            <Button
-              className="pf-m-plain--align-left"
-              variant={ButtonVariant.plain}
-              onClick={handleDeleteProbe}
-            >
-              <MinusCircleIcon />
-            </Button>
-          </Tooltip>
+          <Button variant={ButtonVariant.plain} onClick={showProbe}>
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/components/health-checks/HealthChecksProbe.tsx:64]: Please fix the alignment of the success text

Screenshot from 2020-09-17 14-29-49

**Review by debsmita1 — CHANGES_REQUESTED** [review]

File: frontend/packages/dev-console/src/components/health-checks/HealthChecksProbe.tsx (hunk 2 above)
Comment (sahil143) [reply frontend/packages/dev-console/src/components/health-checks/HealthChecksProbe.tsx]: fixed!

File: frontend/packages/dev-console/src/components/health-checks/health-checks-utils.ts [hunk 3]
```diff
@@ -1,8 +1,15 @@
...
+  K8sResourceKind,
+  ContainerSpec,
+  referenceFor,
+  modelFor,
+} from '@console/internal/module/k8s';
 import * as _ from 'lodash';
 import { getProbesData } from './create-health-checks-probe-utils';
 import { getResourcesType } from '../edit-application/edit-application-utils';
 import { HealthChecksProbeType } from './health-checks-types';
+import { useAccessReview } from '@console/internal/components/utils';
```
Comment (debsmita1) [inline frontend/packages/dev-console/src/components/health-checks/health-checks-utils.ts:12]: please fix the order of this import

Comment (sahil143) [reply frontend/packages/dev-console/src/components/health-checks/health-checks-utils.ts]: fixed

**Comment by debsmita1** [comment]
tried this out locally
/lgtm

**Comment by christianvogt** [comment]
cc @openshift/team-devconsole-ux

**Comment by christianvogt** [comment]
Overall it seems to work fine.
I did notice that when there are no health checks, the disabled action menu now appears in the details page with no actions enabled. Vs before it was simply not present at all.

**Comment by bgliwa01** [comment]
Is there a reason for the the disabled action menu to appear in the details page with no actions enabled?

**Comment by sahil143** [comment]
> Is there a reason for the the disabled action menu to appear in the details page with no actions enabled?

Earlier, we had no action that was enabled in View access so the topology context menu and action menu in sidebar were hidden.

**Pull request merged**